File: frontend/src/types/quizSession.ts
export interface SessionQuestion {
  id: string
  content: string
  question_type: string
  options: string[] | null
  order_index: number
}

export interface QuizSessionCreateResponse {
  session_id: string
  quiz_id: string
  questions: SessionQuestion[]
}

export interface SubmitAnswerResponse {
  answer_id: string
  question_id: string
  is_correct: boolean
  correct_answer: string
  explanation: string | null
  ai_feedback: string | null
}

export interface CompleteQuizSessionResponse {
  session_id: string
  score: number
  total_questions: number
  correct_answers: number
}

export interface ResultAnswer {
  answer_id?: string
  question_id: string
  content: string
  question_type: string
  options: string[] | null
  correct_answer: string
  explanation: string | null
  user_answer: string
  is_correct: boolean
  ai_feedback: string | null
}

export interface QuizSessionResult {
  session_id: string
  quiz_id: string
  quiz_title: string | null
  score: number
  total_questions: number
  correct_answers: number
  started_at: string | null
  completed_at: string | null
  answers: ResultAnswer[]
}

File: backend/app/schemas/quiz_session.py
"""Pydantic schemas for quiz session creation, answers, and results."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class QuizSessionCreateRequest(BaseModel):
    """Payload used to create a quiz session."""

    quiz_id: UUID


class SessionQuestionResponse(BaseModel):
    """Serializable quiz question shown to the learner."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    content: str
    question_type: str
    options: list[str] | None
    order_index: int


class QuizSessionCreateResponse(BaseModel):
    """Response returned when a quiz session starts."""

    session_id: UUID
    quiz_id: UUID
    questions: list[SessionQuestionResponse]


class SubmitAnswerRequest(BaseModel):
    """Payload used to submit one answer for a session question."""

    question_id: UUID
    user_answer: str = Field(min_length=1)


class SubmitAnswerResponse(BaseModel):
    """Evaluation result for one submitted answer."""

    answer_id: UUID
    question_id: UUID
    is_correct: bool
    correct_answer: str
    explanation: str | None
    ai_feedback: str | None


class CompleteQuizSessionResponse(BaseModel):
    """Response returned when finalizing a session."""

    session_id: UUID
    score: float
    total_questions: int
    correct_answers: int


class ResultAnswerResponse(BaseModel):
    """Detailed answer payload shown in the results page."""

    question_id: UUID
    content: str
    question_type: str
    options: list[str] | None
    correct_answer: str
    explanation: str | None
    user_answer: str
    is_correct: bool
    ai_feedback: str | None


class QuizSessionResultResponse(BaseModel):
    """Full session results for the quiz results page."""

    session_id: UUID
    quiz_id: UUID
    quiz_title: str | None
    score: float
    total_questions: int
    correct_answers: int
    started_at: datetime | None
    completed_at: datetime | None
    answers: list[ResultAnswerResponse]
